lib/mongoid/extensions/big_decimal.rb

```diff
@@ -46,7 +50,13 @@ module ClassMethods
         #
         # @return [ BigDecimal, nil ] A BigDecimal derived from the object or nil.
         def demongoize(object)
-          object && object.numeric? ? BigDecimal(object.to_s) : nil
+          if object
```

Comment:
This condition also causes this method to map `false` to `nil`, I suggest `unless object.nil?` instead.

Reply:
False can be ignored as it isn't handled by below statements (it is neither a Decimal128 nor numeric?). Recommend to as a guard method `return unless object`

Reply:
I changed this @p-mongo but I don't think it will make a difference since false is not numeric. I'll leave it because it's more expressive but there's no functionality change. 
```
irb(main):001:0> BigDecimal.demongoize(false)
=> nil
irb(main):002:0> false.numeric?
=> false
```

Reply:
It's true... I guess that's why this PR was cancelled all these years ago...